### DISCUSSION TITLE
use api to call flamegraph?

### DISCUSSION THREAD

**Discussion by zdyj3170101136** [bloomberg/memray#537] [General]
to reduce overhead, i want to dynamic start profile and stop it.

this http request http://:8081/debug/pprof/allocs would start allocation profiling 60 seconds and stop it.

```
from aiohttp import web
import asyncio

prof = Tracer()

async def handle(request):
    asyncio.create_task(heap_profile())
    return web.Response(text="Request received")

async def heap_profile():
    prof.__enter__()
    print("started")
    await asyncio.sleep(60)
    prof.__exit__()
    print("stoped")

app = web.Application()
app.add_routes([web.get('/debug/pprof/allocs', handle)])

web.run_app(app)
```

i know how to produce the bin file by using api.

but how could i use the api to convert bin file to flamegraph?

**Comment by pablogsal** [comment]
> but how could i use the api to convert bin file to flamegraph?

Currently there is no such API. You could shell out to memray to run `memray flamegraph ...` or use `subprocess.run`. We can discuss if we can provide such an API in the future but at the time this is the only way that's supported.
